### DISCUSSION TITLE
My current TODO list

### DISCUSSION THREAD

**Discussion by FlachyJoe** [j8sr0230/fc_nodes#27] [Ideas]
Hi,
FYI I plane to work on these features (unordered):
- [x] refresh nodes list dynamically to take dev into account w/o full reload the app
- [ ] add a workbench parameter page
- [ ] let user open fc_nodes window as FreeCAD tab
- [ ] add nodes category i.e. split the node list
- [ ] add hidden parameters to nodes. Widget without socket or parameter page shown by context menu
- [ ] make a group node (aka. subpatch) node to hide a part of the patch. Based on teleporters
- [ ] Improve node tooltip to show sockets types and docs
- [x] show node icon on its title bar

**Comment by j8sr0230** [comment]
That looks like a plan. We have a small overlap. I am currently working on a way to sort the nodes into categories. I would like to keep it minimalistic and solve everything via a Blender-like context menu (see picture).
![2022-09-14_Context_menu](https://user-images.githubusercontent.com/39261150/190071373-f0b6d570-9c49-43ef-9b59-2c41ae8f41e8.png)
Maybe then we can even remove the node DragList widget. Blender at least gets along without such a feature :) 

Along with your targeted group node, we have a solid node editor to work with, so that we can jump on the nodes. My intention is to publish a structured RoadMap for this purpose, which includes the desired node categories and the nodes to be developed.
The first node categories should be:
- Inputs and outputs,
- Math,
- Lists and vectors,
- Curves,
- Surfaces,
- Solids,
- Mesh,
- ...

What do you think?

**Reply by FlachyJoe** [reply]
Hi !

>I am currently working on a way to sort the nodes into categories.

You're welcome to keep this.

>  Maybe then we can even remove the node DragList widget. Blender at least gets along without such a feature :)

FreeCAD GUI is based on toolbars, I think we should expose nodes in such manner in the workbench in addition to the context menu.
It could be great to implement a search box too.

> What do you think?

It seems realistic :+1:
